dags/multipod/legacy.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run tests migrated from the legacy XL ML infrastructure"""
+
+import datetime
+from airflow import models
+from dags import composer_env, gcs_bucket, test_owner
+from dags.vm_resource import TpuVersion, Zone, Project, DockerImage, ClusterName
+from dags.multipod.configs import legacy_unit_test, gke_config
+from dags.multipod.configs.common import SetupMode, Platform
+
+# Run once a day at 9 am UTC (1 am PST)
+SCHEDULED_TIME = "0 9 * * *" if composer_env.is_prod_env() else None
+
+
+for test_mode in [SetupMode.NIGHTLY, SetupMode.STABLE]:
+  with models.DAG(
```

Comment:
The decorator style might be better here since you have a DAG that takes one argument:

```
@dag(...)
def multipod_legacy_xlml(docker_image):
  ...

multipod_legacy_xlml(DockerImage.MAXTEXT_JAX_NIGHTLY)
multipod_legacy_xlml(DockerImage.MAXTEXT_JAX_STABLE)
```

Reply:
There are a couple of DAG parameters that depend on the test mode (name & tags). I'll see how the refactor looks, but it will probably still need a for loop over setup modes.